File: 2008/J082.pas
var
  n, k, sum, i:longint;
begin
  read(n);
  sum := 0;
  for i := 1 to n do
  begin
    read(k);
    sum := sum + k;
    if (k = n - i - 1) then
    begin
      writeln(n - i - 1);
      break
    end
    else if (k >= n - i) then
    begin
      writeln(n - i);
      break
    end
  end
end.

File: 2008/J082.cpp
#include <cstdio>
int main() {
  int n, k, sum = 0;
  scanf("%d", &n);
  for (int i = 1; i <= n; i++) {
    scanf("%d", &k);
    sum += k;
    if (sum == n - i - 1) {
      printf("%d\n", n - i - 1);
      break;
    } else if (sum >= n - i) {
      printf("%d\n", n - i);
      break;
    }
  }
  return 0;
}
